lib/iris/config.py

```diff
@@ -91,16 +91,25 @@ def get_logger(name, datefmt=None, fmt=None, level=None, propagate=None):
     logger.setLevel(level)
     logger.propagate = propagate
 
-    # Create a formatter.
-    formatter = logging.Formatter(fmt=fmt, datefmt=datefmt)
+    # Create and add the handler to the logger, if required.
+    if handler:
```

Comment:
[The Python logging HOWTO](https://docs.python.org/3/howto/logging.html#loggers) provides an explanation for this change:

>Child loggers propagate messages up to the handlers associated with their ancestor loggers. Because of this, it is unnecessary to define and configure handlers for all the loggers an application uses. It is sufficient to configure handlers for a top-level logger and create child loggers as needed. (You can, however, turn off propagation by setting the _propagate_ attribute of a logger to `False`.)

`handler` therefore needs to be a configurable option so that it can be disabled whenever `propagate` is set to `True`.